rfc6750 compability

hardymansen opened the pull request:
Hi,
Saw https://github.com/auth0/nginx-jwt/issues/43

Support for the WWW-Authenticate response header, according to https://tools.ietf.org/html/rfc6750#section-3.1

For my user case, i need also specific implementation for scope as will be using in OAuth 2.0 framework.
e.g returning a 403, and the scopes required.
Not sure if you are liking that if , or if you want the lib to be more generic.

Mazon committed:
rfc6750 compability

twistedstream commented:
@hardymansen: I think this PR looks good. 
Could you add some test coverage? I think you could simply tweak the existing tests and add some assertions for the new `WWW-Authenticate` header being returned. If not, I could add them as well.

hardymansen commented:
Sure, will have a stab at it.

Mazon committed:
tests cases for rfc6750

Mazon committed:
typos

hardymansen commented:
Added a scope=admin to the default proxy and added the WWW-Authenticate to the test cases, and a new test case for scope missing specific.

Verified and fixed a 200 test. Now all 12 tests pass
![screen shot 2015-12-01 at 18 07 36](https://cloud.githubusercontent.com/assets/8512694/11507933/a7ca98ec-9856-11e5-9ec4-2a80e30e08bd.png)

Mazon committed:
fix 200 test

roryrjb commented:
@twistedstream any update on this?
